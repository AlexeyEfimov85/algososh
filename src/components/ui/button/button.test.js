import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";

import { Button } from "./button";


describe("Button component", () => {
    test("Button без текста рендерится без ошибок", () => {
        render(<Button />);
        const btn = screen.getByRole("button");
        expect(btn).toMatchSnapshot();
      });
      test("Button c текстом рендерится без ошибок", () => {
        render(<Button text={'text'} />);
        const btn = screen.getByRole("button");
        expect(btn).toMatchSnapshot();
      });
      test("Button заблокированная рендерится без ошибок", () => {
        render(<Button disabled={true} />);
        const btn = screen.getByRole("button");
        expect(btn).toMatchSnapshot();
      });
      test("Button с индикацией загрузки рендерится без ошибок", () => {
        render(<Button isLoader={true} />);
        const btn = screen.getByRole("button");
        expect(btn).toMatchSnapshot();
      });
      test ("Нажатие Button вызывает коллбэк", () => {
        window.alert = jest.fn();
        render(<Button onClick={() => alert("коллбэк вызван")} />);
        const btn = screen.getByRole("button");
        fireEvent.click(btn);
        expect(window.alert).toHaveBeenCalledWith("коллбэк вызван");
      })
})

