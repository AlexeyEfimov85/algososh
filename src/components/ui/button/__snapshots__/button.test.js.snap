// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button c текстом рендерится без ошибок 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    text
  </p>
</button>
`;

exports[`Button component Button без текста рендерится без ошибок 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button component Button заблокированная рендерится без ошибок 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button component Button с индикацией загрузки рендерится без ошибок 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;
